### DISCUSSION TITLE
OpenID Certification / Envoy roadmap

### DISCUSSION THREAD

**Discussion by acarlson0000** [antonengelhardt/wasm-oidc-plugin#11] [General]
Hello there @antonengelhardt 

First: thanks for your work on this, much appreciated. My team are currently looking towards OIDC implementation in Envoy, and so coming across this Rust/WASM filter was fantastic - great work so far! Have you spoke to the Envoy team about this work at all to see if there is any alignment?

We'd like the code we're using long term to be certified with OpenID themselves - see https://openid.net/certification/ - have you thought about this and it is part of you approach? 

Pending the above answers, we've noticed that you have a couple of open issues to add further implementation / tests, which is great - are you open to contribution?

Many thanks!

**Comment by antonengelhardt** [comment]
Hey @acarlson0000,

Thank you very much. This project is the result of my bachelor's thesis, which I completed this year. I'm open to contributions and plan to continue working on this project in the future. The issues indicate the next areas of focus.

To answer your other questions:
- No, I haven't spoken to any Envoy team members except Piotr Sikora, the maintainer of [proxy-wasm-rust-sdk](https://github.com/proxy-wasm/proxy-wasm-rust-sdk). The Plugin "framework" is relatively new, and crypto support is [not yet really mature](https://github.com/WebAssembly/wasi-crypto/blob/main/docs/HighLevelGoals.md). I can't provide information about their roadmap for this component.
- I haven't contacted OpenID to certify this work, mainly because I wasn't aware of it until now. Is this part of your compliance requirements?

**Comment by antonengelhardt** [comment]
Hi @acarlson0000,

In the past week, I have done some work on additional features.

If you want to take a look and maybe add some comments about what you think, that would be great.

Also, feel free to open any issues or bugs that you find or would like to see in the future.

Were you also planning to contribute to this repository in the form of pull requests?

Cheers!

**Reply by acarlson0000** [reply]
Hey @antonengelhardt  - Apologies for not getting back to you sooner, we've had various discussion internally within our team around the questions we'd highlighted above, have meant to reply!)

We've been keeping a close eye and very much appreciate the work you've been putting towards this, its fantastic! Yes, I think our plan was to contribute back via the usual PR method as/when we had things to share :)

**Reply by antonengelhardt** [reply]
Ok, sounds great.

I plan to merge #17 soon. Let me know what you think, if you find the time :) @acarlson0000
